Framework/include/QualityControl/BookkeepingQualitySink.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   BookkeepingQualitySink.h
+/// \author Michal Tichak
+///
+
+#ifndef QUALITYCONTROL_BOOKKEEPINGQUALITYSINK_H
+#define QUALITYCONTROL_BOOKKEEPINGQUALITYSINK_H
+
+#include <DataFormatsQualityControl/QualityControlFlagCollection.h>
+#include <Framework/DataProcessorSpec.h>
+#include <Framework/Task.h>
+#include "ProvenanceToQCRunType.h"
+
+namespace o2::quality_control::core
+{
+
+// This class gathers all QualityObjects from it's inputs, converting them to flags + sending them to grpc RCT/BKP when stop() is invoked
+class BookkeepingQualitySink : public framework::Task
+{
+ public:
+  using FlagsMap = std::unordered_map<std::string, std::unique_ptr<QualityControlFlagCollection>>;
```

Comment:
reason why I am using `unordered_map` over `unordered_set` is quite simple: items in map can change, but they cannot change inside the set. So in order to transform code to use set over map, I would need to remove and add flag collection every time it has to be modified. Or create my own implementation. 

So it is good observation, that I basically do not use the key anywhere and it is even possible to change the code to use set, but I wanted to avoid big amount of insert and remove operations in the code.